result/src/main/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -4,59 +4,96 @@ package com.github.kittinunf.result
  * Created by Kittinun Vantasin on 10/26/15.
  */
 
-public interface ResultType<out V : Any, out E : Exception> {
```

Comment:
I do agree that your proposition provide simplicity onto the `Result`.

The reason I designed this into an interface is to provide flexibility for client to conform to. Meaning that, clients can define their own class, implement `ResultType` to get the Result-ish functionality. 

If we do go to the `sealed class` route, @mplatvoet do you have any idea how we can offer some sort of alternative. 

Even better do you feel that aforementioned value is legitimate?